environments/dev-env/CONTRIBUTING.md

```diff
@@ -0,0 +1,510 @@
+# KCP-Edge Contributor Development Environment
+
+## Required Packages:
+   - ko: https://ko.build/install/ 
+   - gcc
+   - docker 
+   - jq
+   - make
+   - go (version expected 1.19)
+   - kind
+   - kubectl  
+
+
+## Supported OS Platforms 
+  - Linux
+  - MacOS
+  - Windows WSL/Ubuntu
+
+
+Run the following script to install the required package (Linux or MacOS ):
+
+```bash
+./install_req.sh
+```
+
+Alternatively, run the following command for MacOS platform:
+
+```bash
+brew install ko gcc jq make go@1.19 kind kubectl
+```
+
+For Windows WSL/Ubuntu platform, follow the instructions [here](docs/wls_ubuntu_platform.md)
+
+
+## Quickstart
+
+### 1. Clone the kcp-edge repo:
+
+```shell
+git clone https://github.com/kcp-dev/edge-mc.git  kcp-edge
+```
+
+### 2. Build the kcp-edge binaries:
+
+```shell
+cd kcp-edge
+make build
+```
+
+The kcp-edge binaries is created inside the `/bin` directory. Add this directory to your $PATH:
+
+```shell
+export PATH="$PATH:$(pwd)/bin"
+```
+
+
+### 3. Create your own edge infrastructure (edge clusters) - Bring Your Own Cluster (BYOC)
+
+Create your edge cluster or bring your own k8s edge cluster. In this example, we use [kind](https://kind.sigs.k8s.io/) to create an edge cluster that we name “florin”:
+
+```shell
+kind create cluster --name florin
+```  
+
+### 4. Deploy the kcp-edge platform:
+
+  a) Download kcp binaries for your platform:
+
+  ```shell
+  mkdir kcp
+  cd kcp
+  VERSION=0.11.0 # choose the latest version (without v prefix)
+  OS=darwin   # linux or darwin
+  ARCH=amd64  # arm64 or amd64
+  curl -sSfL "https://github.com/kcp-dev/kcp/releases/download/v${VERSION}/kcp_${VERSION}_${OS}_${ARCH}.tar.gz" > kcp.tar.gz
+  curl -sSfL "https://github.com/kcp-dev/kcp/releases/download/v${VERSION}/kubectl-kcp-plugin_${VERSION}_${OS}_${ARCH}.tar.gz" > kubectl-kcp-plugin.tar.gz
+  ```
+
+  Extract kcp and kubectl-kcp-plugin and place all the files in the bin directories somewhere in your $PATH. For example:
+
+  ```shell
+  tar -xvf kcp.tar.gz
+  tar -xvf kubectl-kcp-plugin.tar.gz
+  export PATH="$PATH:$(pwd)/bin"
+  ```
+
+  b) Start kcp:
+
+  ```shell
+  kcp start >& kcp_log.txt &
+  ```
+  OR
+  
+  If you need to connect to a remote k8s edge cluster, you can bind kcp to the public interface of your virtual machine:
+
+  ```shell
+  kcp start --bind-address <ip-address> >& kcp_log.txt &
+  ```
+  
+  This uses default 6443 port, just bind to a specific address
```

Comment:
This is confusing as written.  "If you need to connect to a remote k8s edge cluster": why would the server-side address depend on whether the edge cluster is remote?  BTW, the kcp server does not open connections to the edge cluster.  I suspect the issue is about the syncer in the edge cluster being able to open connections to the kcp server.  But I do not understand why specifying a particular server address helps --- the default is to bind to 0.0.0.0, which is the least restrictive thing you can do.

"the public interface of your virtual machine" --- what virtual machine?  No virtual machine has been mentioned, at least not yet.

"This uses default 6443 port, just bind to a specific address" --- the port is potentially important regardless of whether a specific server address is bound.

Reply:
Perhaps the issue is the IP address that kcp writes into the generated kubeconfig.  That is never 0.0.0.0, it picks something more specific.